sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusExtensionConfigProvider.cs

```diff
@@ -108,6 +110,25 @@ public void Initialize(ExtensionConfigContext context)
             context.AddBindingRule<ServiceBusAttribute>().Bind(bindingProvider);
         }
 
+        internal static ParameterBindingData ConvertReceivedMessageToBindingData(ServiceBusReceivedMessage message)
+        {
+            ReadOnlyMemory<byte> messageBytes = message.ToAmqpBytes().ToMemory();
+            byte[] lockTokenBytes = Guid.Parse(message.LockToken).ToByteArray();
+
+            // The lock token is a 16 byte GUID
+            const int lockTokenLength = 16;
+
+            byte[] combinedBytes = new byte[messageBytes.Length + lockTokenLength];
```

Comment:
Is there no way to do buffer rental and returning after the parameter binding data is no longer used?

Reply:
I don't think so - the ownership of the ParameterBindingData is not controlled by the extension.